pkg/server/server.go

```diff
@@ -2055,11 +2081,17 @@ func (s *BgpServer) AddVrf(ctx context.Context, r *api.AddVrfRequest) error {
 			AS:      s.bgpConfig.Global.Config.As,
 			LocalID: net.ParseIP(s.bgpConfig.Global.Config.RouterId).To4(),
 		}
+
 		if pathList, err := s.globalRib.AddVrf(name, id, rd, im, ex, pi); err != nil {
 			return err
 		} else if len(pathList) > 0 {
 			s.propagateUpdate(nil, pathList)
 		}
+		if vrf, ok := s.globalRib.Vrfs[name]; ok {
+			if s.zclient != nil && s.zclient.mplsLabelRangeSize > 0 {
+				go assignMplsLabel(s, vrf)
```

Comment:
why needs to use goroutine here?

Reply:
The function assignMplsLabel waits to read channel (at line 2045 in servr.go).
https://github.com/osrg/gobgp/pull/1992/files#diff-09856fe9becddf0199651f451409356aR2045

The channel is written when zclient receives body of GET_LABEL_CHUNK message (at line 406 in zclient.go)
https://github.com/osrg/gobgp/pull/1992/files#diff-09856fe9becddf0199651f451409356aR2045

I confirmed the program continue to block following processing when I call assignMplsLabel without goroutine.